lib/model.js

```diff
@@ -1412,6 +1412,13 @@ module.exports = (function() {
    * @return {Promise<undefined>}
    */
   Model.prototype.destroy = function(options) {
+    var self = this
+      , instances;
+
+    if (!options || !options.where) {
```

Comment:
https://github.com/sequelize/sequelize/blob/master/lib/dialects/abstract/query-generator.js
https://github.com/sequelize/sequelize/blob/master/lib/dialects/postgres/query-generator.js#L405

We do support truncate down the line, so you might want to add a check and test to make sure that `truncate: true` will still work correctly.